bundle/DependencyInjection/Compiler/WebpFormatVariationPathGeneratorDecoratorPass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\SiteBundle\DependencyInjection\Compiler;
+
+use Netgen\Bundle\SiteBundle\Core\Imagine\VariationPathGenerator\WebpFormatVariationPathGenerator;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class WebpFormatVariationPathGeneratorDecoratorPass implements CompilerPassInterface
```

Comment:
Do we really need the compiler pass just to register a decorator?

Can't we do this through the service config in YAML files?

Reply:
Half of my life you force me to write compiler passes, now you want me to add it in the config 😆 
I wasn't initially sure if we want to make the decorator service enabled by default, or if we will need to introduce a switch through configuration. @vjeran wants this enabled by default in both legacy admin and symfony/ezplatform, so maybe no need for the compiler pass.

Reply:
Compiler passes should be used when modifying eixsting services! With this, we're not modifying an existing service, but adding a new one.

Besides, we're already using decorators directly in yaml files.

Reply:
Let's leave it like this for consistency sake, as we will be using the compiler pass to actually decorate the existing variation path generator in Ibexa 4.x with our own service. If we are using 4.x in combination with legacy admin, then this is needed to consolidate variation path patterns to match the ones from legacy. The compiler pass in that case will replace ibexa decorator class with our own.